hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/HoodieTimelineArchiver.java

```diff
@@ -504,24 +505,27 @@ private Stream<HoodieInstant> getInstantsToArchive() {
           .setBasePath(HoodieTableMetadata.getDatasetBasePath(config.getBasePath()))
           .setConf(metaClient.getHadoopConf())
           .build();
-      Option<HoodieInstant> earliestActiveDatasetCommit = dataMetaClient.getActiveTimeline().firstInstant();
-
-      if (config.shouldArchiveBeyondSavepoint()) {
```

Comment:
why we have removed the code path for config.shouldArchiveBeyondSavepoint(). don't we need special handling ?

Reply:
Right.  The savepoints need to be handled separately.  Fixed.